### PR TITLE
Implement `Interop.HRESULT`

### PR DESCRIPTION
**Pull request by xtqqczze**
cc: @daxian-dbw

### CODE REVIEW

**Comment by xtqqczze** [comment]
> @xtqqczze Could you please add a reference to source code of the `ComHresults.IsSuccess`?

updated description

**Comment by xtqqczze** [comment]
`src\System.Management.Automation\engine\ComInterop\ComHresults.cs` is excluded from compilation :(

https://github.com/PowerShell/PowerShell/blob/f943426f99cddca3506764445a07c78d1f633189/src/System.Management.Automation/System.Management.Automation.csproj#L69

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by xtqqczze** [comment]
@daxian-dbw Ready for review, with an added remark on the provenance of `ComHresults`.

File: src/System.Management.Automation/ComInterop/ComHresults.cs [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 namespace System.Management.Automation.ComInterop
 {
+    /// <remarks>
+    /// This class is derived from
+    /// <see href="https://github.com/dotnet/runtime/blob/1f688ed75133664aa20c838fa451e2ecf9f1f424/src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/ComHresults.cs">dotnet/runtime</see>.
+    /// </remarks>
```
Comment (daxian-dbw) [inline src/System.Management.Automation/ComInterop/ComHresults.cs:9]: There is no need to add this remark. The code in `ComInterop` folder are all ported from .NET.

Comment (xtqqczze) [reply src/System.Management.Automation/ComInterop/ComHresults.cs:9]: Although we have `src/System.Management.Automation/engine/ComInterop/README.md`, this PR moves `ComHresults.cs` into a new `ComInterop` directory in `src/System.Management.Automation`. 

I also think documenting the provenance to a specific commit is worthwhile.

Comment (daxian-dbw) [reply src/System.Management.Automation/ComInterop/ComHresults.cs:9]: Ah, I didn't notice you moved the file. Please don't move the file.
We try to keep the `ComInterop` as intact as possible so in future if we need to port new changes to COM support from .NET, it's easier to do diff.

If moving the file is required for this cleanup to work, then I'd rather to continue to use `Utils.Succeeded`.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/Interop/Interop.HRESULT.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
```
Comment (daxian-dbw) [inline src/System.Management.Automation/Interop/Interop.HRESULT.cs:4]: I don't see a reason for making it more complex. I would prefer to keep the `Utils.Succeed`

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/Interop/Interop.HRESULT.cs (hunk 2 above)
Comment (iSazonov) [reply src/System.Management.Automation/Interop/Interop.HRESULT.cs:4]: I don't see too. 😕

**Comment by xtqqczze** [comment]
Closing this PR as no benefit for now.

In the medium-term, we should move to using CsWin32 source generator, which implements `Succeeded` in the same way as this PR:
https://github.com/microsoft/CsWin32/blob/main/src/Microsoft.Windows.CsWin32/templates/HRESULT.cs

**Pull request closed without merging**